src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamespaceSymbolExtensions.cs

```diff
@@ -26,4 +26,129 @@ public static IEnumerable<INamedTypeSymbol> GetNamedTypeSymbols(this INamespaceS
 			}
 		}
 	}
-}
\ No newline at end of file
+	
+	public static string GetFullTypeString(this INamedTypeSymbol type) => 
+		type.Name + 
+		(type.NullableAnnotation == NullableAnnotation.Annotated ? "?" : string.Empty) +
+		type.TypeArguments.GetGenericTypeArgumentsString();
+
+	public static string GetGenericTypeArgumentsString(this IEnumerable<ITypeSymbol> typeArguments)
+	{
+		string result = string.Empty;
+
+		if (!typeArguments.Any())
+		{
+			return result;
+		}
+
+		result += "<";
+
+		bool isFirstArgument = true;
+		foreach (ITypeSymbol typeArg in typeArguments)
+		{
+			if (isFirstArgument)
+			{
+				isFirstArgument = false;
+			}
+			else
+			{
+				result += ", ";
+			}
+
+			switch (typeArg)
+			{
+				case ITypeParameterSymbol typeParameterSymbol:
+					result += typeParameterSymbol.Name;
+					break;
+				case INamedTypeSymbol namedTypeSymbol:
+					result += namedTypeSymbol.GetFullTypeString();
+					break;
+				default:
+					//Maybe should handle this somehow, maybe inject SourceProductionContext and call ReportDiagnostic on it?
```

Comment:
Going to `default` will cause what?

Reply:
I suppose other derivatives of `ITypeSymbol` must be handled here, but that is my first experience with roslyn and I'm not sure if they can appear in this loop, maybe if one will define `TArrayOfTypes[]` as a generic type parameter or in some other case

Reply:
I couldn't identify any case where default would happen, if you can point me at possible condition let me know